WordPress/src/main/java/org/wordpress/android/ui/domains/management/MyDomainsScreen.kt

```diff
@@ -0,0 +1,105 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.network.rest.wpcom.site.AllDomainsDomain
+import org.wordpress.android.ui.compose.components.MainTopAppBar
+import org.wordpress.android.ui.compose.components.NavigationIcons
+import java.time.Instant
+import java.time.LocalDate
+
+@Composable
+fun MyDomainsScreen(uiState: DomainManagementViewModel.UiState) {
+    Scaffold(
+        topBar = {
+            MainTopAppBar(
+                title = "My Domains",
+                navigationIcon = NavigationIcons.BackIcon,
+                onNavigationIconClick = {},
+            )
+        },
+    ) {
+        Column (
+            modifier = Modifier.padding(it),
+        ){
+            MyDomainsSearchInput()
+            when (uiState) {
+                is DomainManagementViewModel.UiState.Populated ->
+                    MyDomainsList(uiState.domains)
+                else -> {}
+            }
+        }
+    }
+}
+
+@Composable
+fun MyDomainsSearchInput() {
+    var queryString by rememberSaveable { mutableStateOf("") }
+
+    OutlinedTextField(
+        value = queryString,
+        onValueChange = { queryString = it },
+        placeholder = { Text("Search your domains") },
+        shape = RoundedCornerShape(50),
+        leadingIcon = {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_search_white_24dp),
+                contentDescription = "",
+                tint = MaterialTheme.colorScheme.outline,
+            )
+        },
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(16.dp)
+    )
+}
+
+
+@Composable
+fun MyDomainsList(domains: List<AllDomainsDomain>) {
+    LazyColumn(
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+        modifier = Modifier.padding(16.dp),
+    ) {
+        items(items = domains) {
+            DomainListCard(
+                uiState = DomainCardUiState(
+                    domain = it.domain,
+                    title = null,
+                    status = it.domainStatus,
+                    expiry = it.expiry?.time?.let { epoch ->
```

Comment:
I got an error in the `LocalDate` conversion and tried out a [different conversion from here](https://github.com/wordpress-mobile/WordPress-Android/blob/57a414e6b95a5216cdbb32909208daa1a384b921/WordPress/src/main/java/org/wordpress/android/ui/jetpackoverlay/JetpackFeatureRemovalBrandingUtil.kt#L98) that worked for me.
```
FATAL EXCEPTION: main
Process: com.jetpack.android.beta, PID: 18535
j$.time.DateTimeException: Unable to obtain LocalDate from TemporalAccessor: 2024-03-24T00:00:00Z of type j$.time.Instant
```

For reference my test data are the following:
```
{
  "domains": [
    {
      "domain": "antonis.me",
      "blog_id": 58348,
      "blog_name": "antonis.log()",
      "type": "mapping",
      "is_domain_only_site": false,
      "is_wpcom_staging_domain": false,
      "has_registration": false,
      "registration_date": "2009-03-26T21:20:53+00:00",
      "expiry": "2024-03-24T00:00:00+00:00",
      "wpcom_domain": false,
      "current_user_is_owner": true,
      "site_slug": "antonis.me",
      "domain_status": {
        "status": "Active",
        "status_type": "success"
      }
    },
    {
      "domain": "antonistest1.blog",
      "blog_id": 217792462,
      "blog_name": "Site Title",
      "type": "mapping",
      "is_domain_only_site": false,
      "is_wpcom_staging_domain": false,
      "has_registration": false,
      "registration_date": "1970-01-01T00:00:00+00:00",
      "expiry": false,
      "wpcom_domain": false,
      "current_user_is_owner": false,
      "site_slug": "antonistest1.wordpress.com",
      "domain_status": {
        "status": "Expiring soon",
        "status_type": "error",
        "status_weight": 1000
      }
    }
  ]
}
```

Reply:
Thanks for the suggestion Antonis! I updated this with https://github.com/wordpress-mobile/WordPress-Android/pull/19366/commits/59df94b93802acf371f2f17ac458bd7aa540e634, otherwise, the suggestions on GitHub would have created two commits.